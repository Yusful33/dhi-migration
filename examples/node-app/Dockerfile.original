# Base image
FROM node:20.19.2-alpine3.21

# Set the user to root (default in official Node image, but weâ€™ll be explicit)
USER root

# Install some useful packages
RUN apt-get update && \
    apt-get install -y --no-install-recommends \
        curl \
        vim \
        jq \
    && rm -rf /var/lib/apt/lists/*

# Set working directory
WORKDIR /app

# Copy application files
COPY package*.json ./
RUN npm install
COPY . .

# Expose a port under 1080 (e.g., 80)
EXPOSE 80

# Start the application (assumes it binds to port 80 inside the app)
CMD ["node", "server.js"]
