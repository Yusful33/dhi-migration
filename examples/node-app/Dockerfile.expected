# Dockerfile migrated to Docker Hardened Images (DHI)
# Generated by DHI Migration Tool

# Build stage - using DHI dev image with build tools
FROM demonstrationorg/dhi-node:20.19.2-alpine3.21-dev AS build-stage
WORKDIR /app
COPY package*.json ./
RUN npm ci --omit=dev
COPY . .

# Runtime stage - using minimal DHI image
FROM demonstrationorg/dhi-node:20.19.2-alpine3.21 AS runtime-stage
WORKDIR /app
COPY --from=build-stage --chown=nonroot:nonroot /app /app
EXPOSE 3000
CMD ["npm", "start"]